x/stablestake/keeper/msg_server_add_pool.go

```diff
@@ -0,0 +1,47 @@
+package keeper
+
+import (
+	"context"
+	"errors"
+
+	errorsmod "cosmossdk.io/errors"
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/elys-network/elys/x/stablestake/types"
+)
+
+func (k msgServer) AddPool(goCtx context.Context, msg *types.MsgAddPool) (*types.MsgAddPoolResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	params := k.ammKeeper.GetParams(ctx)
+
+	if !params.IsCreatorAllowed(msg.Sender) {
+		return nil, errors.New("sender is not allowed to create pool")
+	}
+
+	found := k.HasPoolByDenom(ctx, msg.DepositDenom)
+	if found {
+		return nil, errorsmod.Wrapf(types.ErrPoolAlreadyExists, "pool with denom %s already exists", msg.DepositDenom)
+	}
+
+	poolId := k.GetNextPoolId(ctx)
+	pool := types.Pool{
+		PoolId:               poolId,
+		DepositDenom:         msg.DepositDenom,
```

Comment:
lets have a check which ensures it's not base currency for the pool

Reply:
it's not base currency for pool ? could you please add detail ?